File: src/algorithms/missing-number.js
/**
 * Given an array nums containing n distinct numbers in the range [0, n],
 * returns the only number in the range that is missing from the array.
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number} - Returns the missing number from range
 */
function missingNumber (nums) {
  nums.sort((a, b) => {
    return a - b;
  });
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== i) {
      return i;
    }
  }
  return nums.length;
}

File: src/algorithms/missing-number.py
def missing_number(nums):
    """Given an array nums containing n distinct numbers in the range [0, n], 
    return the only number in the range that is missing from the array.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns the integer missing from the range.
    """
    nums = sorted(nums)
    size = len(nums)
    if nums[0] != 0:
        return 0
    if nums[size - 1] != size:
        return size
    for i in range(size):
        if nums[i] != i:
            return i
    return 0
